greeter/README.md

```diff
@@ -1,21 +1,706 @@
+   this.add(createBindingFromClass(CachingInterceptor));
+
+   this.component(RestExplorerComponent);
+   this.component(GreetingComponent);
+   ```
+
+### Step 4: Try it out again!
+
+Let's try it out to see the caching server and the observer that we just bind. Restart your application.
+
+#### Test cache
+
+The expire time for each result is set to 10s (10000 milliseconds by default).
+To test the cached result and its expiration, you can
+
+Try GET/ greet/{name} with a name 'loopback' and language 'en'
+You will see a result object with a timestamp field like
+
+```s
+{
+  "timestamp": "2019-10-09T20:24:00.294Z",
+  "language": "en",
+  "greeting": "Hello, loopback!"
+}
+```
+
+Try the same endpoint and same input immediately (before the cached result expires), you will see the same timestamp printed in the result, which infers that the result is NOT calculated again but from cache.
+Wait for 10s and try same endpoint again, you will see the timestamp changes, like (maybe use an image to point out the difference)
+
+```s
+{
+// please note the time here changes
+"timestamp": "2019-10-09T20:30:29.770Z",
+"language": "en",
+"greeting": "Hello, loopback!"
+}
+```
+
+#### Test sweep
+
+The sweep interval is set to 10s(10,000 milliseconds) too.
+Since the cache sweep runs every 10s in the background, we can observe the sweep function gets executed by turning on the debug string and restart the app.
+
+To test the cache sweeping, you can:
+
+Stop your app and run it again with debug string turning on:
+
+```s
+// exit your app
+ctrl + c
+// start it again with debug
+DEBUG=greeter-extension npm run start
+```
+
+Watch the console window to see how the sweeping function is triggered every 10s
+
+```s
+greeter-extension Sweeping cache... +10s
+greeter-extension Sweeping cache... +10s
+greeter-extension Sweeping cache... +10s
+greeter-extension Sweeping cache... +10s
+greeter-extension Sweeping cache... +10s
+```
+
+Try endpoint GET /greet/{name} again from explorer, see its result
+Go back to the console window, after 10s, you will see the cached result gets swept
+
+```s
+Cache for en:/greet/loopback is swept.
+```
+
+Summary
+The interceptor, service and observer could be contributed by from different modules, or packaged together as a component, all depends on your choice. You just bind them to the app without modifying the business logic of your endpoint.
+
+Back to the concepts, now audience should have a more through understanding of how to "provide aspect-oriented logic around method invocations" and how to "allow artifacts to participate in the life cycles".
+
+## Conclusion
+
+Congratulations! You have completed the tutorial.
+
+You have successfully built a LoopBack application using an existing extension `greeter-extension`, and then you have added functionality on top of the existing extension, i.e. add a French greeter. Lastly, you have enabled caching as part of the application lifecycle using a lifecycle observer.
```

Comment:
> You have successfully built a LoopBack application using an existing extension `greeter-extension`, and then you have added functionality on top of the existing extension, i.e. add a French greeter. Lastly, you have enabled caching as part of the application lifecycle using a lifecycle observer. 


First, you built a LoopBack application using a component `GreetingComponent` located in a different module named  `greeter-extension`. Then you extended the functionality to the  `GreetingComponent` by adding `FrenchGreeter` as an extension. Then you created a caching service that is started and stopped by a lifecycle observer. And lastly, you retrieved/stored request responses from/in the cache by using an interceptor.